component/attr_cache/cacheMap.go

```diff
@@ -58,6 +65,14 @@ type cacheTreeMap struct {
 	cacheTree *attrCacheItem
 }
 
+// type passed to insert
+type insertOptions struct {
```

Comment:
What are the main benefits to supplying options into the insert? So far it seems more readable to see what parameters are being set to what values. Does this add more flexibility to how insert can be used? if so, how?

Reply:
I chose to add a struct because I wanted to add a parameter (`fromDirList`) with a default value (`false`).
The struct allows me to do that, since Go does not have optional function parameters like Python does.
Another slight benefit is that filling out the `insertOptions` struct when calling `insert` makes each parameter's name visible, so there's no question why we're passing the literal `true` in as the second parameter.
The downside, of course, is how verbose it is. It seems like a lot of noise to add so many lines and words.
I'm not sure if this was the right decision, and I'm eager to hear what it's like to read this. Is it noisy? Readable? Annoying?